_posts/2023-02-13-changelog-188.markdown

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #188"
+date: 2023-02-13 13:00:00 +0300
+---
+
+
+## New Features
+
+* Detect new compiler errors (by [@kuksag]):
+  * [#10066] `main` function is not allowed to have generic parameters ([E0131](https://doc.rust-lang.org/error-index.html#E0131))
+
+  * [#10066] An inherent implementation was marked `unsafe` ([E0197](https://doc.rust-lang.org/error-index.html#E0197))
+
+  * [#10066] Type parameter has more than one relaxed default bound ([E0203](https://doc.rust-lang.org/error-index.html#E0203))
+
+  * [#9894] A pattern was declared as an argument in a foreign function declaration ([E0130](https://doc.rust-lang.org/error-index.html#E0130))
+
+  * [#9894] Trait methods currently cannot take patterns as arguments ([E0642](https://doc.rust-lang.org/error-index.html#E0642))
+
+* [#10057] Parse and annotate [`const_closures`](https://github.com/rust-lang/rust/issues/106003) syntax
+
+* [#8826] Support new `where` clause location for associated types
+
+* [#9987] Annotate missing lifetimes ([E0106](https://doc.rust-lang.org/error-index.html#E0106)) in function signatures
+
+* [#9123] Allow excluding specific paths from auto-import via `Preferences | Editor | General | Auto Import | Rust` settings.
+By default, [`Borrow`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) and [`BorrowMut`](https://doc.rust-lang.org/std/borrow/trait.BorrowMut.html) methods are excluded, as auto-import is rarely desired for them.
+Tip: exclude functions or traits that are re-exported in irrelevant crates to get only the original one in the suggestions.
+For further details, please refer to the pull request description
+
+  <img src="/assets/posts/changelog-188/exclude_paths_auto_import.png" width="600px"/>
+
+* [#7489] Add debugger option to not step into stdlib sources (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-188/do_not_step_into_stdlib" w="600px" %}
+
+* [#9962] Show the contents of raw slice pointers (`*const [T]`, `*mut [T]`) in the debugger
+
+  <img src="/assets/posts/changelog-188/contents_of_slice_ptr.png" width="600px"/>
+
+* [#9806] Update icons for the [New UI](https://www.jetbrains.com/help/idea/new-ui.html)
+
+  <img src="/assets/posts/changelog-188/icons_new_ui.png" width="600px"/>
+
+* [#10056] Add a dedicated icon for procedural macros
+
+  <img src="/assets/posts/changelog-188/icon_proc_macro.png" width="600px"/>
+
+* [#10050] Allow disabling gutter icons provided by the plugin via `Preferences | Editor | General | Gutter Icons` settings
+
+* [#9583] Support `Emulate terminal in output console` option for `wasm-pack` [configurations](https://plugins.jetbrains.com/plugin/8182-rust/docs/wasm-projects-support.html#wasm-pack-config)
+
+## Performance Improvements
+
+* [#9972] Use mixed AST where possible
```

Comment:
It's unclear for users why it affects performance

Reply:
Moved to `Internal Improvements` based on the PR description